com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/TextureGenerator.java

```diff
@@ -560,6 +477,65 @@ public static TextureImage generate(BufferedImage origImage, TextureProfile texP
+            return TextureImage.CompressionType.COMPRESSION_TYPE_BASIS_UASTC;
+        }
+        return defaultType;
+    }
+
+    private static TextureImage.TextureFormat textureFormatToSupportedTextureFormat(TextureImage.TextureFormat format) {
+        // Issue 5753: Since we currently don't support pre compressed hardware textures, so we use UASTC instead
+        if (format == TextureFormat.TEXTURE_FORMAT_RGB_PVRTC_2BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGB_PVRTC_4BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGB_ETC1) {
+            return TextureFormat.TEXTURE_FORMAT_RGB;
+        } else if (format == TextureFormat.TEXTURE_FORMAT_RGBA_PVRTC_2BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGBA_PVRTC_4BPPV1) {
+            return TextureFormat.TEXTURE_FORMAT_RGBA;
+        }
+        return format;
+    }
+
+    private static String compressionTypeToTextureCompressor(TextureImage.CompressionType type) {
+        return switch (type) {
+            case COMPRESSION_TYPE_DEFAULT,
+                 COMPRESSION_TYPE_WEBP -> TextureCompressorDefault.TextureCompressorName;
+            case COMPRESSION_TYPE_BASIS_ETC1S,
+                 COMPRESSION_TYPE_BASIS_UASTC,
+                 COMPRESSION_TYPE_WEBP_LOSSY -> TextureCompressorBasisU.TextureCompressorName;
+        };
+    }
+
+    private static String compressionLevelToTextureCompressorPreset(TextureImage.CompressionType type, TextureFormatAlternative.CompressionLevel level) {
+        // Convert from basis to basis preset
+        if (type == TextureImage.CompressionType.COMPRESSION_TYPE_BASIS_UASTC || type == TextureImage.CompressionType.COMPRESSION_TYPE_BASIS_ETC1S) {
+            if (level == TextureFormatAlternative.CompressionLevel.FAST) {
+                return "BASISU_FAST";
+            } else if (level == TextureFormatAlternative.CompressionLevel.NORMAL) {
+                return "BASISU_NORMAL";
+            } else if (level == TextureFormatAlternative.CompressionLevel.HIGH) {
+                return "BASISU_HIGH";
+            } else if (level == TextureFormatAlternative.CompressionLevel.BEST) {
+                return "BASISU_BEST";
+            }
+        }
+        else if (type == TextureImage.CompressionType.COMPRESSION_TYPE_DEFAULT) {
+            return "DEFAULT";
+        }
+        return null;
+    }
+
+    private static TextureImage.CompressionType textureCompressorToCompressionType(String compressor) {
+        if (compressor.equals(TextureCompressorDefault.TextureCompressorName)) {
+            return TextureImage.CompressionType.COMPRESSION_TYPE_DEFAULT;
+        } else if (compressor.equals(TextureCompressorBasisU.TextureCompressorName)) {
+            return TextureImage.CompressionType.COMPRESSION_TYPE_BASIS_UASTC;
+        }
+        // TODO: This shouldn't be needed eventually, but right now we need a compression type in the engine.
+        return TextureImage.CompressionType.COMPRESSION_TYPE_DEFAULT;
```

Comment:
This is referring to that we currently don't have a way to plugin in a compressor/decoder in the engine for extensions.